client/app/components/tags-control/EditTagsDialog.jsx

```diff
@@ -1,66 +1,70 @@
 import { map, trim, uniq, compact } from "lodash";
-import React from "react";
+import React, { useState, useEffect } from "react";
 import PropTypes from "prop-types";
 import Select from "antd/lib/select";
 import Modal from "antd/lib/modal";
 import { wrap as wrapDialog, DialogPropType } from "@/components/DialogWrapper";
 
-class EditTagsDialog extends React.Component {
-  static propTypes = {
-    dialog: DialogPropType.isRequired,
-    tags: PropTypes.arrayOf(PropTypes.string),
-    getAvailableTags: PropTypes.func.isRequired,
-  };
+function EditTagsDialog({ dialog, tags, getAvailableTags }) {
+  const [availableTags, setAvailableTags] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [values, setValues] = useState(() => uniq(map(tags, trim))); // lazy evaluate
+  const [selectRef, setSelectRef] = useState(null);
```

Comment:
Lastly, is there a reason for this to be a State instead of a `useRef`?

Reply:
Yes, to be able to use it as a dependency for `useEffect`. When `ref.current` changes, component won't re-render, and I'd like to avoid race condition here - when request completes before ref initializes. I'm not sure how it may be possible, but if it happens - it will be a hard-to-find bug.

Reply:
Nice, I figured this could be the case, thanks for explaining 🙂.